content/2. Ondersteunende-informatie/4. Techniek/Git/7. GitHub Actions.md

```diff
@@ -0,0 +1,125 @@
+## Wat zijn GitHub Actions?
+
+GitHub Actions is een CI/CD-tool ([Uitleg over CI/CD](https://www.redhat.com/en/topics/devops/what-is-ci-cd)) die het mogelijk maakt om automatisch workflows uit te voeren in reactie op gebeurtenissen in een GitHub-repository, zoals push- of pull-request-activiteiten. Met GitHub Actions kun je eenvoudig geautomatiseerde workflows opzetten voor builds, tests, en deployments.
+
+### Structuur van een Workflow
+Een workflow wordt gedefinieerd in een `.yml`-bestand dat zich in de map `.github/workflows/` bevindt. De basisstructuur van een workflowbestand is als volgt:
+
+```
+name: Example name for a workflow
+
+on:
+  event_type: 
+    branches:
+      - main  
+
+jobs:  
+  naam-van-een-job: 
+    runs-on: ubuntu-latest  
+    steps:  
+      - name: Checkout code  
+        uses: actions/checkout@v2  
+      
+      - name: Run a script
+        run: |
+          echo "Hello, World!"
+
+```
+
+### Uitleg van de commando's
+```
+name: Example name for a workflow
+```
+- `name`: De naam van de workflow die zichtbaar is in GitHub. Het is handig om hier een korte duidelijke beschrijving staat wat de workflow doet.
+
+```
+on:
+  push:  
+    branches:
+      - main  
+```
+- `on`: Dit specificeert wat er moet gebeuren om de workflow te laten runnen
+- `push`: Dit specificeert op basis van welke actie de workflow getriggerd wordt.
+	Dit kunnen ook andere event types zijn:
+	- `pull`
+	- `commit`
+	- `pull_request`
+	- `release`
+	- `schedule`
+	- `workflow_dispatch`
+- `branches`: Dit specificeert op welke branch de actie moet zijn uitgevoerd om de workflow te triggeren.
+
+```
+jobs:  
+  naam-van-een-job: 
+    runs-on: ubuntu-latest  
+    steps:  
+      - name: Checkout code  
+        uses: actions/checkout@v2  
+      
+      - name: Run a script
+        run: |
+          echo "Hello, World!"
+```
+- `jobs`: Dit kan een lijst zijn van alle verschillende taken die uitgevoerd moeten worden. In dit voorbeeld is er 1 job.
+- `naam-van-een-job`: De naam van een job die zichtbaar is in GitHub. Deze naam kan geen spaties bevatten, alleen letters, `-` en `_`.
+- `runs-on`: Specificeert op welk besturingssysteem de job moet werken.
+- `steps`: Dit is een lijst van de stappen die de action op volgorde zal doorlopen. Elke stap heeft weer een eigen vorm
+	- `name`: De naam van de stap die zichtbaar is op GitHub.
+	- `uses`:  De actie die de stap moet uitvoeren op basis van al bestaande actions. Dit kunnen eigen actions zijn op actions van andere mensen.
+	- `run: |`: Dit kan allerlei verschillende commando's uitvoeren van bijvoorbeeld een terminal, python of npm.
+
+### Overige commando's 
+```
+on:
+  push: 
+    branches:
+      - main 
+    paths-ignore:
+	  - `.github/**`  
+```
+- `paths-ignore`: Dit zegt tegen de action dat de content in de folder van `.github` moet negeren.
+- De tegenhanger hiervan, `paths`,  zorgt ervoor dat de action dus alleen op die folders let.
+
```

Comment:
Nog toevoegen wat de on: push branches: main dan precies doet. In normaal NL dan. Dus iets als: "In dit geval wordt deze action uitgevoerd als er in de main branch wordt gepusht." of iets vergelijkbaars